lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -9,39 +9,66 @@
 	/**
 	 * Whether a theme or its parent have a theme.json file.
 	 *
-	 * @param boolean $clear_cache Whether the cache should be cleared and theme support recomputed. Default is false.
+	 * The result would be cached via the WP_Object_Cache.
+	 * It can be cleared by calling wp_theme_has_theme_json_clean_cache().
 	 *
 	 * @return boolean
 	 */
-	function wp_theme_has_theme_json( $clear_cache = false ) {
-		static $theme_has_support = null;
+	function wp_theme_has_theme_json() {
+		$cache_group       = 'theme_json';
```

Comment:
Once in core, I think we should move [this](https://github.com/spacedmonkey/wordpress-develop/commit/b10228fc4aa106438e0a989f30b7340973d8f786). This would bring it inline with the `wp_is_block_theme` function.